benchexec/resources.py

```diff
@@ -209,145 +170,191 @@ def _get_cpu_cores_per_run0(
                 f"of core {core} are not usable. "
                 f"Please always make all virtual cores of a physical core available."
             )
+    # check if all units of the same hierarchy level have the same number of cores
+    for index in range(len(hierarchy_levels)): # [dict, dict, dict, ...]
```

Comment:
It seems here to use a for loop with a loop counter, but you only use the loop counter to access `hierarchy_levels[index]`. So you could also simply use a `for hierarchy_level in hierarchy_levels:` loop. In general, the latter style is preferred and should be always used when possible.

Reply:
Edited loop and corresponding asymmetry check in https://github.com/sosy-lab/benchexec/pull/892/commits/943aaebd910dd09583a0f5b44b0d4bd75b30fae2